File: share-code/bai-tap-lap-trinh/python/shape/14.py
num_rows = 5
result = ""

for i in range(1, num_rows + 1):
    space = " " * (num_rows - i)
    right = ""

    for j in range(1, i + 1):
        right += str(j)

    left = right[::-1][:-1]

    result += space + left + right + "\n"

print(result, end="")

File: share-code/bai-tap-lap-trinh/cpp/05-thaphinh/14.cpp
#include <iostream>
#include "zvn.h"

using namespace std;

int main()
{
    /*
       1        1
      212       2
     32123      32
    4321234
   543212345
    */
    int number = 1;
    const int HEIGHT = 5;
    string shape = "";

    for (int i = number; i <= HEIGHT; i++)
    {
        string space = myRepeat(" ", HEIGHT - i);
        string left = createListNumberReverse(i, 2);
        string right = createListNumber(2, i);
        string record = space + left + "1" + right + space;
        shape += record + "\n";
    }
    cout << shape << "\n";
}